wrangler/src/@utils/auth.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Helper functions for managing authentication against a Turret
+ */
+import BigNumber from 'bignumber.js'
+import moment from 'moment'
+import { Transaction, Networks } from 'stellar-base'
+import { verifyTxSignedBy } from './stellar-sdk-utils'
+
+/**
+ * Determine the authenticated user and contracts with the provided token.
+ * 
+ * @param {string} network The Stellar Network to validate the token on
+ * @param {string} authToken - The provided auth token
+ * @param {string?} dataKey - (Optional) Fetch all ManageData operations such that the key matches this dataKey.
+ *                           All ManageData values whose keys match will be included in the data output array.
+ * @link https://tyvdh.github.io/stellar-tss/#section/Authentication
+ * 
+ *
+ * @typedef {Object} AuthResult
+ * @property {string} publicKey - The authenticated public key from the token.
+ * @property {string[]} data - The data collected. Returns an empty array if no matching ManageData values are 
+ *                             found or no dataKey was provided.
+ *
+ * @returns {AuthResult} - The result of the authentication
+ */
+export function authTxToken(network, authToken, dataKey) {
+  try {
+    if (authToken === undefined) {
+      throw { message: 'Unable to find an auth token' };
```

Comment:
As a rule of thumb, it's always better to use `Error` constructor (e.g. `new Error('Unable to find an auth token')` as it provides a built-in error stacktrace and better supported by other language primitives like `async` and Promises.

Reply:
I think this is an expected format of the `ctx` cloudflare utils that this repo uses to format responses? I am not positive on this fact, so going to leave as is unless @tyvdh can say yes or no.

Reply:
We need to go through the entire service and redo how errors and captured and reported. Stack traces being a good reason for that. It's really hard to decipher where issues come from atm. This is work for another PR though as this is the pattern atm.